### PR TITLE
added function to add missing values in to ts in order regularlize se…

### PR DESCRIPTION
**Pull request by mbannert**
experimental for series that can be regularized using yearly freq as a basis.

### CODE REVIEW

File: R/regularize.R [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#' dv <- c(seq(as.Date("2010-01-01"), length = 3, by="3 years"),
+#' seq(as.Date("2018-01-01"), length = 2, by="2 years"))
+#' library(zoo)
+#' xx <- zoo(ts1,dv)
+#' regularize(xx)
+#' @export
+regularize <- function(x, df = "%Y"){
+  reg <- as.numeric(format(index(x), df))
+  full_r <- seq(range(reg)[1],range(reg)[2])
+  full_r[!(full_r %in% reg)]
```
Comment (HomoCodens) [inline R/regularize.R]: I assume this is a leftover line that is not necessary for the function to work? :wink:

**Comment by HomoCodens** [comment]
Neat!

Though I think rather than add such a function specifically as a fix for the case with yearly series that sparked it, we should try to support the most common frequencies used with tstools.

`as.numeric(format(index(x), "%Y"))` is only going to work for yearly data I believe, as `format(index(x), "%Y-%m")` results in "2010-01" for example, which will result in an `NA` after `as.numeric`.

I tried coming up with a more general variant:
```R
regularize2 <- function(x) {
  t <- index(x)
  dt <- as.numeric(diff(t))/365
  freqs <- c(12, 4, 2, 1)
  frq <- freqs[which.min(abs(min(dt) - 1/freqs))]
  full_r <- seq(min(t), max(t), by = sprintf("%d months", 12/frq))
  val <- rep(NA, length(full_r))
  val[(full_r %in% t)] <- x
  tx <- ts(val, start = as.yearmon(min(t)), freq = frq)
  tx
}

ts1 <- rnorm(5)
dv <- c(seq(as.Date("2010-01-01"), length = 3, by="3 years"),
        seq(as.Date("2018-01-01"), length = 2, by="2 years"))
library(zoo)
xx <- zoo(ts1,dv)
regularize2(xx)

dv2 <- c(seq(as.Date("2010-01-01"), length = 20, by = "6 months"))
dv2 <- dv2[c(1:10, 14:20)]
xx2 <- zoo(rnorm(length(dv2)), dv2)
regularize2(xx2)
```

Do you think this might work? For frequencies other than `c(12, 4, 2, 1)` it will still not work of course.

**Comment by mbannert** [comment]
I agree. Code seems to work for me. Do you see any potential leap year issues?

**Comment by HomoCodens** [comment]
I don't think so. The difference between a normal year and a leap year is `1/365` or `~0.0027` so that should not cause any problems when determining the closest `1/freqs`.

```r
  freqs <- c(12, 4, 2, 1)
  frq <- freqs[which.min(abs(min(dt) - 1/freqs))]
```

**Pull request merged**